backend/app/routers/experiments.py

```diff
@@ -133,93 +132,99 @@ async def delete_experiment(
     await Experiment.find(Experiment.id == id).delete()
 
 
 @router.patch("/experiments/{id}/archive", response_model=None)
 async def archive_experiment(
     id: PydanticObjectId,
-    is_archived: bool = False,
+    archived: bool = False,
     user: dict = Depends(get_current_user(required=True)),
 ) -> Any:
     experiment = await get_experiment_if_accessible_or_raise(
         id, user, write_access=True
     )
-    experiment.is_archived = is_archived
+    experiment.archived = archived
 
     await Experiment.replace(experiment)
 
 
 def find_specific_experiments(
     search_query: str,
-    only_mine: bool,
-    only_not_archived: bool,
-    only_public: bool,
-    query_operator: QueryOperator,
+    mine: bool | None,
+    archived: bool | None,
+    public: bool | None,
     user: dict | None,
+    query_operator: QueryOperator = QueryOperator.AND,
     pagination: Pagination = None,
 ) -> FindMany[Experiment]:
-    search_conditions = []
     page_kwargs = (
         {"skip": pagination.offset, "limit": pagination.limit}
         if pagination is not None
         else {}
     )
+    # initial condition -> retrieve only those objects that are accessible to a user
+    search_conditions = [
+        operators.Or(
+            ExperimentTemplate.public == True,  # noqa: E712
+            Eq(
+                ExperimentTemplate.created_by, user["email"] if user is not None else ""
+            ),
+        )
+    ]
+
+    # applying filters
     if len(search_query) > 0:
         search_conditions.append(Text(search_query))
-
-    if only_mine:
+    if mine is not None:
         if user is not None:
-            search_conditions.append(Experiment.created_by == user["email"])
+            search_conditions.append(
+                get_compare_operator_fn(mine)(Experiment.created_by, user["email"])
+            )
         else:
             # Authentication required to see your experiment templates
             raise HTTPException(
                 status_code=status.HTTP_401_UNAUTHORIZED,
                 detail="This endpoint requires authorization. You need to be logged in.",
                 headers={"WWW-Authenticate": "Bearer"},
             )
-
-    if only_not_archived:
-        search_conditions.append(Experiment.is_archived == False)  # noqa: E712
-    if only_public:
-        search_conditions.append(Experiment.is_public == True)  # noqa: E712
-
-    if len(search_conditions) > 0:
-        multi_query = (
-            operators.Or(*search_conditions)
-            if query_operator == QueryOperator.OR
-            else operators.And(*search_conditions)
-        )
-        return Experiment.find(multi_query, **page_kwargs)
-
-    return Experiment.find_all(**page_kwargs)
+    if archived:
+        search_conditions.append(Experiment.archived == archived)  # noqa: E712
+    if public:
+        search_conditions.append(Experiment.public == public)  # noqa: E712
+
+    multi_query = (
+        operators.Or(*search_conditions)
```

Comment:
The initial condition regarding accessibility should be always required. We can not concatenate it with `OR` like this, I assume.